streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordQueue.java

```diff
@@ -92,7 +93,14 @@ int addRawRecords(final Iterable<ConsumerRecord<byte[], byte[]>> rawRecords) {
                 continue;
             }
 
-            final long timestamp = timestampExtractor.extract(record, timeTracker.get());
+            final long timestamp;
+            try {
+                timestamp = timestampExtractor.extract(record, timeTracker.get());
+            } catch (final StreamsException internalFatalExtractorException) {
+                throw internalFatalExtractorException;
+            } catch (final Exception fatalUserException) {
+                throw new StreamsException("Fatal user code error in TimestampExtractor callback.", fatalUserException);
```

Comment:
In this case should we add the partition/offset etc?

Reply:
?

Reply:
Ack.